components/StarSearch/StarSearchCompactHeader.tsx

```diff
@@ -0,0 +1,49 @@
+import { ArrowLeftIcon, HistoryIcon, PlusIcon, XIcon } from "@primer/octicons-react";
+import { HiOutlineShare } from "react-icons/hi";
+import { StarSearchLogo } from "./StarSearchLogo";
+
+type OnClick = (e: React.MouseEvent<HTMLButtonElement>) => void;
+
+interface StarSearchCompactHeaderProps {
+  onBack: OnClick;
+  onShare: OnClick;
+  onNewChat: OnClick;
+  onShowHistory: OnClick;
+  onClose: OnClick;
+}
+
+export const StarSearchCompactHeader = ({
+  onBack,
+  onShare,
+  onNewChat,
+  onShowHistory,
+  onClose,
+}: StarSearchCompactHeaderProps) => {
+  const buttonHoverStyle = "hover:text-orange-500";
+
+  return (
+    <div className="flex items-center justify-between gap-2 [&_button]:text-slate-600">
+      <div className="flex items-center gap-2">
+        <button onClick={onBack} className={buttonHoverStyle}>
+          <ArrowLeftIcon />
+        </button>
+        <StarSearchLogo size="small" />
+        <span className="font-semibold text-slate-800">StarSearch</span>
```

Comment:
Figma says Slate 800 which I put, but should this just be our default text colour from the body @isabensusan? They seem very similar.